x-pack/plugins/security_solution_serverless/public/navigation/index.ts

```diff
@@ -32,16 +30,21 @@ export const startNavigation = (services: Services) => {
 
   enableManagementCardsLanding(services);
 
-  const projectNavigationTree = new ProjectNavigationTree(services);
-
   if (services.experimentalFeatures.platformNavEnabled) {
-    const SideNavComponentWithServices = withServicesProvider(SecuritySideNavComponent, services);
-    serverless.setSideNavComponent(SideNavComponentWithServices);
+    initProjectNavigation(services).then(
+      ({ navigationTree$, panelContentProvider, dataTestSubj }) => {
+        serverless.initNavigation(navigationTree$, { panelContentProvider, dataTestSubj });
+      }
+    );
```

Comment:
Would it be possible to keep this `setupNavigation` function declarative, moving the async logic inside the `initProjectNavigation` function? so this line would be just:

```suggestion
    initProjectNavigation(services);
```

Reply:
The `initProjectNavigation` would look like
```
export const initProjectNavigation = (services: Services) => {
  const { serverless } = services;
  import('./project_navigation').then(({ init }) => {
    const { navigationTree$, panelContentProvider, dataTestSubj } = init(services);
    serverless.initNavigation(navigationTree$, { panelContentProvider, dataTestSubj });
  });
};
```
The rest of the changes look good 👍

Reply:
Addressed in https://github.com/elastic/kibana/pull/174544/commits/320ef4eea2a102db630e33d2eb366be471ec3512